File: Assets/Scripts/Proto/AckScenePlayers.cs
using System.Collections;
using System.Collections.Generic;


public class AckScenePlayers
{
	private List<MsgScenePlayer> _players = new List<MsgScenePlayer>();


	public AckScenePlayers(Packet packet)
	{
		this._players = new List<MsgScenePlayer>();
		ushort players_count = packet.ReadUshort();
		for (ushort i = 0; i < players_count; i++)
		{
			this._players.Add(new MsgScenePlayer(packet));
		}
	}


	public List<MsgScenePlayer> players
	{
		get { return this._players; }
		set { this._players = value; }
	}

}

File: pytools/examples/java/client/protocol/AckScenePlayers.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class AckScenePlayers
{
	private List<Msgsceneplayer> players = new ArrayList<Msgsceneplayer>();


	public AckScenePlayers(Packet packet)
	{
		int players_count = PacketUtil.readUShort(packet.readShort());
		for (int i = 0; i < players_count; i++)
		{
			players.add(new MsgScenePlayer(packet));
		}
	}



	public List<Msgsceneplayer> getPlayers()
	{
		return this.players;
	}

}
